packages/admin/cms-admin/src/generator/future/generator.ts

```diff
@@ -57,10 +58,20 @@ export type TabsConfig = { type: "tabs"; tabs: { name: string; content: Generato
 
 type DataGridSettings = Pick<GridColDef, "headerName" | "width" | "minWidth" | "maxWidth" | "flex">;
 
+export type StatusMapping = Record<
+    string,
+    {
+        color: SvgIconTypeMap["props"]["color"];
```

Comment:
I think it would be better to allow any icon here than only `<StateFilled color="..." />`. For instance, in the page tree we have a combination of different icons and colors for the page visibility component, which is basically a status.

Reply:
I was planning to add this ability in a follow up PR. 
Wasn't sure how the generator-config and the import of the icons should work, so I just used `StateFilled` for now, so we at least have something useful relatively quickly. 

There is also a separate task for this: SVK-316

Reply:
What do you think about adding an optional `icon` key (component-name as a string) here to the config object?

```tsx
statusMapping: {
    value: { icon: "Favorite", color: "success", primaryText: "In Stock" },
    // ...
},
```

Generally I would prefer a ReactNode but I'm not sure if this can be done in an intuitive way, as the import from the config would have to be used in the generator, the ReactNode would have to be copied by the generator as a string... 

```tsx
statusMapping: {
    value: { primaryText: "In Stock", icon: <Favorite color="success" /> },
    // ...
},

Reply:
We don't have a solution for imports in the config (yet), but using ReactNode would be the best

Reply:
> What do you think about adding an optional icon key (component-name as a string) here to the config object?

This would require all icons to come from a predefined package (`@comet/admin-icons`, presumably), so I'm not sure if this is the best solution.

Reply:
Then I'll do the component-name/string for now. 
We can change this, once we have a solution.